Enable TLSv1.3 by default

mattcaswell opened the pull request:
<!--
Thank you for your pull request. Please review these requirements:

Contributors guide: https://github.com/openssl/openssl/blob/master/CONTRIBUTING

Other than that, provide a description above this comment if there isn't one already

If this fixes a github issue, make sure to have a line saying 'Fixes #XXXX' (without quotes) in the commit message.
-->

##### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->
- [x] documentation is added or updated
- [x] tests are added or updated

mattcaswell commented:
Updated with a change to travis.yml (no need to specify enable-tls1_3 any more). Also added extended tests to commit message.

levitte commented on CHANGES:
```diff
@@ -9,6 +9,26 @@
+     eventually published. Different pre-release versions may implement
+     different versions of the draft. The final version of OpenSSL 1.1.1 will
+     implement the final version of the standard.
+     TODO(TLS1.3): Remove the above note before final release
```
Add something in NEWS as well

levitte commented:
[This job](https://travis-ci.org/openssl/openssl/jobs/338119153) has found an issue, it seems:

```
../test/recipes/70-test_clienthello.t .............. 
1..1
    # Subtest: ../../test/clienthellotest
    1..1
        # Subtest: test_client_hello
        1..4
        ok 1 - iteration 1
        ok 2 - iteration 2
        ok 3 - iteration 3
        # ERROR: (bool) 'msglen == F5_WORKAROUND_MAX_MSG_LEN == true' failed @ test/clienthellotest.c:203
        # false
        not ok 4 - iteration 4
    not ok 1 - test_client_hello
../../util/shlib_wrap.sh ../../test/clienthellotest ../../test/session.pem => 1
not ok 1 - running clienthellotest

#   Failed test 'running clienthellotest'
#   at ../test/recipes/70-test_clienthello.t line 20.
# Looks like you failed 1 test of 1.
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/1 subtests 
```

mattcaswell commented:
Added a fixup commit to add something to NEWS. Also added a new commit to fix the clienthello test issue. Basically the clienthello test checks that the padding extension is working properly. This requires that the ClientHello size is kept within certain bounds. With TLSv1.3 enabled by default, and combined with other options used in the travis tests, the ClientHello size got to be too big.

mattcaswell commented:
> CIs seem happy and I'm happy

I think you spoke too soon! Both travis and appveyor failed.

Travis because of some undefined behaviour that came to light in the ossltest engine. Appveyor
because tls13encryptiontest is an "internal" test. As with all of the internal tests this is not supported
on native windows when built "shared". But the test recipe did not skip the execution in this case.

New commits pushed addressing the above. @levitte please reconfirm?

mattcaswell commented:
Pushed. Thanks!

FdaSilvaYY commented on INSTALL:
```diff
@@ -482,27 +482,24 @@
-                   versions.
+                   versions. Note that there is no "no-tls1_3-method" option
+                   because there is no application method for TLSv1.3. Using
+                   invidivial protocol methods directly is deprecated.
```
individual

mattcaswell replied:
#5329
